lms/djangoapps/grades/rest_api/v1/views.py

```diff
@@ -207,3 +219,318 @@ def _get_course(self, request, course_id):
+
+            * page_size: Number of results to return per page. Optional.
+
+        **Response Values**
+
+            If the request for information about the course grade status is successful, an HTTP 200 "OK" response
+            is returned.
+
+            The HTTP 200 response has the following values.
+
+            * results: A list of the course grading status matching the request.
+
+                * course_id: Course ID of the course in the course grading status.
+
+                * user: Username of the user in the course enrollment.
+
+                * passed: Boolean flag for user passing the course.
+
+                * current_grade: An integer represnting current grade of the course.
+
+                * section_breakdown: This is a list of dictionaries which provide details on sections
+                    that were graded.
+
+                A dictionary in the section_breakdown list has the following keys:
+                    * percent: A float percentage for the section.
+                    * label: A short string identifying the section. Preferably fixed-length. E.g. "HW  3".
+                    * detail: A string explanation of the score. E.g. "Homework 1 - Ohms Law - 83% (5/6)"
+                    * category: A string identifying the category.
+                    * prominent: A boolean value indicating that this section should be displayed as more prominent
+                        than other items.
+
+            * next: The URL to the next page of results, or null if this is the
+                last page.
+
+            * previous: The URL to the next page of results, or null if this
+                is the first page.
+
+            If the user is not logged in, a 401 error is returned.
+
+            If the user is not global staff, a 403 error is returned.
+
+            If the specified course_id is not valid or any of the specified usernames
+            are not valid, a 400 error is returned.
+
+            If the specified course_id does not correspond to a valid course or if all the specified
+            usernames do not correspond to valid users, an HTTP 200 "OK" response is returned with an
+            empty 'results' field.
+        """
+        course_grading_status = []
+        course_keys = []
+        username_filter = None
+
+        form = CourseEnrollmentsApiListForm(self.request.query_params)
+        if not form.is_valid():
+            raise ValidationError(form.errors)
+        usernames = form.cleaned_data.get('username')
+        course_id = form.cleaned_data.get('course_id')
+        username_filter = []
+        if usernames:
+            username_filter = [Q(user__username__in=usernames)]
+        course_enrollments = self._paginate_course_enrollment(course_id, course_enrollment_filter=username_filter)
+        course_keys = SectionGradesBreakdown._get_enrollment_course_keys(course_enrollments)
+        users = GradeViewMixin._get_enrolled_users(course_enrollments)
```

Comment:
We need to do something like the following snippet. Otherwise, paginating by users AND courses will lead to duplicates. See my example from [this comment](https://github.com/openedx/edx-platform/pull/31491#discussion_r1101595264). In that example, `User 2`'s enrollment in Demo1 will be listed on pages 1 and 2.

If users are enrolled in multiple courses, they can be present on multiple paginated pages.

```suggestion
        users = defaultdict(list)
        for enrollment in course_enrollments:
            users[enrollment.course_id].append(enrollment.user)
```

I verified this locally, and on my devstack (with different courses and enrollments) `page_size=6` returns 11 results. `page_size=10` returns 59 results.

Reply:
Thanks for pointing this out